fineract-provider/src/main/java/org/apache/fineract/infrastructure/security/data/TwoFactorConfigurationValidator.java

```diff
@@ -0,0 +1,120 @@
+
+    private final FromJsonHelper fromJsonHelper;
+
+    @Autowired
+    public TwoFactorConfigurationValidator(FromJsonHelper fromJsonHelper) {
+        this.fromJsonHelper = fromJsonHelper;
+    }
+
+    public void validateForUpdate(final String json) {
+        if (StringUtils.isBlank(json)) {
+            throw new InvalidJsonException();
+        }
+
+        boolean atLeastOneParameterPassedForUpdate = false;
+        final Type typeOfMap = new TypeToken<Map<String, Object>>() {}.getType();
+        this.fromJsonHelper.checkForUnsupportedParameters(typeOfMap, json,
+                TwoFactorConfigurationConstants.REQUEST_DATA_PARAMETERS);
+        final JsonElement element = this.fromJsonHelper.parse(json);
+
+        final List<ApiParameterError> dataValidationErrors = new ArrayList<>();
+        final DataValidatorBuilder baseDataValidator = new DataValidatorBuilder(dataValidationErrors)
+                .resource(TwoFactorConfigurationConstants.RESOURCE_NAME);
+
+
+        for(String parameterName : TwoFactorConfigurationConstants.BOOLEAN_PARAMETERS) {
+            if(this.fromJsonHelper.parameterExists(parameterName, element)) {
+                atLeastOneParameterPassedForUpdate = true;
+                validateBooleanParameter(parameterName, element, baseDataValidator);
+            }
+        }
+
+        for(String parameterName : TwoFactorConfigurationConstants.STRING_PARAMETERS) {
+            if(this.fromJsonHelper.parameterExists(parameterName, element)) {
+                atLeastOneParameterPassedForUpdate = true;
+                validateStringParameter(parameterName, element, baseDataValidator);
+            }
+        }
+
+        for(String parameterName : TwoFactorConfigurationConstants.NUMBER_PARAMETERS) {
+            if (this.fromJsonHelper.parameterExists(parameterName, element)) {
+                atLeastOneParameterPassedForUpdate = true;
+                validateNumberParameter(parameterName, element, baseDataValidator);
+            }
+        }
+
+        if(!atLeastOneParameterPassedForUpdate) {
+            final Object forceError = null;
+            baseDataValidator.reset().anyOfNotNull(forceError);
+        }
+
+        throwExceptionIfValidationWarningsExist(dataValidationErrors);
+    }
+
+    private void throwExceptionIfValidationWarningsExist(
+            final List<ApiParameterError> dataValidationErrors) {
+        if(!dataValidationErrors.isEmpty()) {
+            throw new PlatformApiDataValidationException(dataValidationErrors);
+        }
+    }
+
+    private void validateBooleanParameter(final String name, final JsonElement element,
+                  final DataValidatorBuilder baseDataValidator) {
+        final String value = this.fromJsonHelper.extractStringNamed(name, element);
+        baseDataValidator.reset().parameter(name).value(value).notNull().trueOrFalseRequired(value);
+    }
+
+    private void validateStringParameter(final String name, final JsonElement element,
```

Comment:
with this approach of validation, we can not validate on length property.
if for any parameter if have max length 50 , then it will hit to database.